serrano/resources/stats.py

```diff
@@ -7,9 +8,39 @@
 from avocado.query import pipeline
 from avocado.core.cache import cache_key
 from avocado.core.cache.model import NEVER_EXPIRE
+from ..conf import settings
 from .base import BaseResource, ThrottledResource
 
 
+def get_count(request, model, refresh, processor, context):
+    opts = model._meta
+
+    # Build a queryset through the context which is toggled by
+    # the parameter.
+    processor = processor(context=context, tree=model)
+    queryset = processor.get_queryset(request=request)
+
+    # Get count from cache or database
+    label = ':'.join([opts.app_label, opts.module_name, 'count'])
+    key = cache_key(label, kwargs={'queryset': queryset})
+
+    if refresh:
+        count = None
+    else:
+        count = cache.get(key)
+
+    if count is None:
+        count = queryset.values('pk').distinct().count()
+        cache.set(key, count, timeout=NEVER_EXPIRE)
+
+    # Close the connection in the thread to prevent 'idle in transaction'
+    # situtations.
+    from django.db import connection
+    connection.close()
```

Comment:
I don't particularly like this, nor can I say I know all of the implications of this, but in the test suite connections would *hang on* even though the threads are cleaned up.